Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -0,0 +1,562 @@
+import macro from 'vtk.js/Sources/macro';
+import vtkAbstractWidgetFactory from 'vtk.js/Sources/Widgets/Core/AbstractWidgetFactory';
+import vtkPlane from 'vtk.js/Sources/Common/DataModel/Plane';
+import vtkPlaneSource from 'vtk.js/Sources/Filters/Sources/PlaneSource';
+import vtkResliceCursorContextRepresentation from 'vtk.js/Sources/Widgets/Representations/ResliceCursorContextRepresentation';
+
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+
+import widgetBehavior from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior';
+import stateGenerator from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/state';
+
+import {
+  getClosestPoint,
+  boundPoint,
+  updateState,
+} from 'vtk.js/Sources/Widgets/Widgets3D/ResliceCursorWidget/helpers';
+import { ViewTypes } from 'vtk.js/Sources/Widgets/Core/WidgetManager/Constants';
+
+import { vec4, mat4 } from 'gl-matrix';
+
+const VTK_INT_MAX = 2147483647;
+const { vtkErrorMacro } = macro;
+
+// ----------------------------------------------------------------------------
+// Factory
+// ----------------------------------------------------------------------------
+
+function vtkResliceCursorWidget(publicAPI, model) {
+  model.classHierarchy.push('vtkResliceCursorWidget');
+
+  model.methodsToLink = ['activeColor', 'useActiveColor', 'opacity'];
+
+  // --------------------------------------------------------------------------
+  // Private methods
+  // --------------------------------------------------------------------------
+
+  function updatePlanePoints(line1, line2, origin, planeSource) {
+    const point1 = getClosestPoint(origin, [
+      line1.getPoint1(),
+      line1.getPoint2(),
+    ]);
+    const point2 = getClosestPoint(origin, [
+      line2.getPoint1(),
+      line2.getPoint2(),
+    ]);
+    planeSource.setPoint1(point2);
+    planeSource.setPoint2(point1);
+  }
+
+  function computeReslicePlaneOrigin(viewType) {
```

Comment:
let's try to keep the use of `viewType` as limited as possible. Use `[plane]normal`/orientation` instead.

Reply:
Not sure to understand whyyou want to use the orientation. The goal of this method is to setup a basic plane with bounded points. Then, we apply the effective normal to the plane, which will update the point.

Reply:
I don't like `viewType` very much (a bit hack-ish), therefore I want to make sure it is used as few as possible.
If viewType was replaced by slicingMode (0, 1, 2 for i, j, k), you could then factorize setOrigin, setPoint1 and setPoint2. e.g.
```
const origin = [
  bounds[0] + offset[0],
  bounds[2] + offset[1],
 bounds[2] + offset[1]
];
origin[slicingMode] = center[slicingMode];
```
Now, if you have slicingMode working, you can simply support the more generic version:
```
const mainOrientation = normal.reduce((max, component) => Math.max(max, component);
slicingMode = normal.indexOf(mainOrientation);
```

Reply:
I don't understand the issue. Why is it not a good idea to use it here ? The viewType is only use in this file.
Do you mean that if I don't use viewType, then in function updateReslicePlane, I'll have to convert viewType into an int (0, 1 or 2 ) ? Which I think, is a little bit more confusing if we want to follow the workflow. I'd like to have a readable code and I'm afraid that this solution complicate the readability of the code (conversion between number and explicit enum)

Reply:
I do not like the new ViewTypes enum values. However I have nothing better to suggest. So leave it as is